src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -123,55 +149,50 @@ internal static bool IsGloballyEnabled()
                     base.Send(request, cancellationToken);
                 return response;
             }
-            catch (OperationCanceledException)
+            catch (Exception ex) when (LogException(ex, request, ref taskStatus))
```

Comment:
Why use "when" here? I find it confusing.

Reply:
I think using `when` just for logging came from https://github.com/dotnet/runtime/pull/37619#discussion_r441867363.

We use it in [HttpMessageInvoker](https://github.com/dotnet/runtime/blob/4fd380a71666afb8572b5e6559cafcbd72a469b9/src/libraries/System.Net.Http/src/System/Net/Http/HttpMessageInvoker.cs#L94), [Dns](https://github.com/dotnet/runtime/blob/4fd380a71666afb8572b5e6559cafcbd72a469b9/src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs#L28) and [SslStream](https://github.com/dotnet/runtime/blob/4fd380a71666afb8572b5e6559cafcbd72a469b9/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs#L264) because of that.

I don't have a strong preference here, but I assume we'd want to be consistent across the places we do this. That said I suspect we are already inconsistent in a bunch of places.

Reply:
I hadn't seen this before, thanks.

I understand the rationale re debuggability.

Still, seems like a weird pattern to me...

Reply:
Reverted the `when` here.